src/jotai.ts

```diff
@@ -32,21 +32,30 @@ export function atom<AtomType>(
         computeValue();
       });
     }
-    
+
     return currentValue;
   }
 
+  function getInitialType() {
+    const _type = typeof initialValue;
+    if (initialValue?.then && typeof initialValue.then === "function") {
+      return "promise";
+    }
+    return _type;
+  }
+
   async function computeValue() {
     const newValue =
       typeof initialValue === "function"
         ? (initialValue as AtomGetter<AtomType>)(get)
         : value;
-    value = (null as AtomType);
+
+    value = null as AtomType;
     value = await newValue;
     subscribers.forEach((callback) => callback(value));
   }
 
-  computeValue();
+  if (["promise", "function"].includes(getInitialType())) computeValue();
```

Comment:
without this modifycation the code below breaks in testing

const basicAtom = atom(2)
expect(basicAtom.get()).toEqual(2)

so we can assume we only want to call computed if we are dealing with a promise or function. promise though?